src/utils.jl

```diff
@@ -27,3 +27,42 @@ function contiguous_regions(condition::AbstractVector{Bool})
     # Reshape the result into two columns
     return reshape(idx,2,length(idx) ÷ 2)'
 end
+#UTIL function used in recpies
+"""
+merge_overlapping_gtis(gtis::Matrix{Float64})::Matrix{Float64}
+Merge overlapping Good Time Intervals (GTIs).
+This function takes a matrix of GTIs, where each row represents a start and stop time,
+and returns a new matrix with non-overlapping GTIs.
+The input matrix `gtis` should have two columns, where the first column is the start time and the second column is the stop time.
+If the input matrix has one or zero rows, it is returned unchanged.
+The function sorts the GTIs by start time, then iterates through them, merging any overlapping intervals.
+The output is a matrix with the same two-column format, containing the merged GTIs.
+"""
+function merge_overlapping_gtis(gtis::Matrix{Float64})::Matrix{Float64}
+    if size(gtis, 1) <= 1
+        return gtis
+    end
+    sort_indices = sortperm(gtis[:, 1])
+    sorted_gtis = gtis[sort_indices, :]
```

Comment:
You can likely use `Base.permutecols!` or similar for this to avoid creating a copy of all of the GTIs.